aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -366,6 +366,31 @@ Tensor& logical_not_out(Tensor& result, const Tensor& self) {
   return result;
 }
 
+Tensor& signbit_out(Tensor& result, const Tensor& self) {
+  TORCH_CHECK(!self.is_complex(), "clamp is not yet implemented for complex tensors.");
+  result.resize_(self.sizes());
+
+  if (self.dtype() == at::kBool) {
+    return result.fill_(false);
+  } else {
+    TensorIterator iter = TensorIteratorConfig()
+      .check_all_same_dtype(false)
+      .set_check_mem_overlap(true)
+      .add_output(result)
+      .add_input(self)
+      .promote_inputs_to_common_dtype(true)
+      .cast_common_dtype_to_outputs(true)
```

Comment:
This doesn't need to promote inputs or cast to an output. There's only one input and for this first iteration signbit doesn't need to support non-bool outputs.

Reply:
Got it! thanks for your explanation.